### PR TITLE
ADEN-5963 Adds MR unblocking and ability to disable SRA (wf ADEN-5887)

### PR DESCRIPTION
**Pull request by leshniak**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/885928d9ae646f9620bf9f68bfb62c8bfdefe38e/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/885928d9ae646f9620bf9f68bfb62c8bfdefe38e/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by fraszczakszymon — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/885928d9ae646f9620bf9f68bfb62c8bfdefe38e/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/3e8fa5a95672d1473d4d2345130182fd04160f4c/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by fraszczakszymon — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/a13488de6f59b0b712f73fd4b236d2170c2df299/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/9d067e35200ab3b69ad2a1f2f468a5cc888f412a/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/provider/gpt/helper.js [hunk 1]
```diff
@@ -97,6 +98,10 @@ define('ext.wikia.adEngine.provider.gpt.helper', [
 			});
 		}
 
+		if (extra.sraEnabled) {
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/provider/gpt/helper.js]: Do we need to pass it through method calls?
Maybe we can have this flag in `adContext`?

File: extensions/wikia/AdEngine/js/provider/directGpt.js [hunk 2]
```diff
@@ -23,7 +23,17 @@ define('ext.wikia.adEngine.provider.directGpt', [
 ) {
 	'use strict';
 
-	var context = adContext.getContext();
+	var context = adContext.getContext(),
+		sraEnabled = !context.opts.disableSra,
+		atfSlots = [
+			'INVISIBLE_SKIN',
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/provider/directGpt.js]: Can we remove skin slot when SRA is disabled?

File: extensions/wikia/AdEngine/js/provider/gpt/helper.js [hunk 3]
```diff
@@ -210,6 +210,10 @@ define('ext.wikia.adEngine.provider.gpt.helper', [
 		}
 	}
 
+	function isSraEnabled() {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/provider/gpt/helper.js]: MAJOR Remove unused function 'isSraEnabled'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/39e59864a7e5b230b39f61c4a8a304e2199a91b2/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/23561bd1052cc980f7aa5d773584bad62f3ee1b2/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/77b7987b2ed0f927ab1a9ff48cc6e02aab9a2bd9/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by kvas-damian — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L78](https://github.com/Wikia/app/blob/edf2b9e243d4a8e6cb68102a53e99e39bb2e7195/extensions/wikia/AdEngine/js/context/adContext.js#L78): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [adContext.js#L86](https://github.com/Wikia/app/blob/edf2b9e243d4a8e6cb68102a53e99e39bb2e7195/extensions/wikia/AdEngine/js/context/adContext.js#L86): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**